app/Http/Controllers/CustomerController.php

```diff
@@ -42,21 +42,27 @@ public function create(PostCodeRequest $request)
         $url = 'https://zipcloud.ibsnet.co.jp/api/search?zipcode=' . $zipcode;
 
         $client = new Client();
-        try {
-            $response = $client->request($method, $url);
-            $body = $response->getBody();
-            $zip_cloud = json_decode($body, true);
+        $response = $client->request($method, $url);
+        $body = $response->getBody();
+        $zip_cloud = json_decode($body, true);
+
+        switch($zip_cloud['status']){
+        case 400:
+            return view('/customers/post_code', ['message' => $zip_cloud['message']]);
+            break;
+        case 500:
+            return view('/customers/post_code', ['message' => $zip_cloud['message']]);
+            break;
+        case 200:
             $result = $zip_cloud['results'][0];
             $address = $result['address1'].$result['address2'].$result['address3'];
             $post_code = $result['zipcode'];
-        }catch(\Throwable $th){
-            $address = null;
+            return view('customers.create')->with(compact('address', 'post_code'));
+            break;
         }
```

Comment:
この場合はcaseをまとめることもできるはずです😊

```suggestion
        switch($zip_cloud['status']){
        case [400, 500]:
            return view('/customers/post_code', ['message' => $zip_cloud['message']]);
            break;
        case 200:
            $result = $zip_cloud['results'][0];
            $address = $result['address1'].$result['address2'].$result['address3'];
            $post_code = $result['zipcode'];
            return view('customers.create')->with(compact('address', 'post_code'));
            break;
        }
```

200以外の動作は同じなら以下の方法もありますね😁

```suggestion
        if($zip_cloud['status'] == 200){
            $result = $zip_cloud['results'][0];
            $address = $result['address1'].$result['address2'].$result['address3'];
            $post_code = $result['zipcode'];
        } else {
            return view('customers.create')->with(compact('address', 'post_code'));
        }
```

他にもアドレスが複数設定されている場合とかもあるので考えてみると面白いと思います😇

Reply:
こんなに色々な書き方が！
ありがとうございます！
私の修正に関しては全然後回しで大丈夫です、ありがとうございました！！